src/borg/repository.py

```diff
@@ -1319,28 +1319,48 @@ def _close_fd(self, ts_fd):
         safe_fadvise(fd.fileno(), 0, 0, 'DONTNEED')
         fd.close()
 
+    def get_segment_dirs(self, data_dir, start_index=0, end_index=2**30):
+        """Returns generator yielding required segment dirs in data_dir as `os.DirEntry` objects.
+        Start and end are inclusive.
+        """
+        segment_dirs = (
+            f
+            for f in os.scandir(data_dir)
+            if f.is_dir() and f.name.isdigit() and start_index <= int(f.name) <= end_index
+        )
+        return segment_dirs
+
+    def get_segment_files(self, segment_dir, start_index=0, end_index=2**32):
```

Comment:
let's use `2**32-1`, see line 1346.

also change get_segment_dirs in the same way.

Reply:
maybe we could have `constants.MIN_SEGMENT = 0` and `.MAX_SEGMENT = 2**32-1` and use them in repository.py everywhere.

Reply:
Is there a possibility that the starting segment (~0) might change? 
Otherwise, could we just store the upper bounds as constants, since starting at 0 is pretty self-explanatory.

Reply:
Well, likely not, but I'ld say if we introduce constants, we do it for both min and max.

Reply:
Done